### PR TITLE
GitUtil: Add try-catch block when resolving Git repo

### PR DESCRIPTION
**Pull request by wong-codaio**
**GitUtil: Add try-catch block when resolving Git repo**
* Added simple try-catch around `simpleGit()` initialization - If anything error'ed out, `null` will be returned.

### CODE REVIEW

File: js/src/gitutil.ts [hunk 1]
```diff
@@ -18,10 +18,16 @@ export interface RepoStatus {
...
+  try {
+    const git = simpleGit();
+    if (await git.checkIsRepo()) {
+      return git;
+    } else {
+      return null;
+    }
+  }
+  catch  (e) {
+    console.warn('Error occurred while checking if current directory is a git repo.', e);
```
Comment (wong-codaio) [inline js/src/gitutil.ts:30]: @ankrgyl - Not sure if this will be too noisy.  Can also drop the logging if it's not helpful.

Comment (ankrgyl) [reply js/src/gitutil.ts:30]: I think it's ok to drop, to be consistent with the Python implementation. We can always add it in to debug further if needed.

Out of curiosity, what did it print in your use case?

**Review by ankrgyl — COMMENTED** [review]
Code LGTM. Thank you!!

Just running the tests on our end to make sure it doesn't break anythign

**Comment by ankrgyl** [comment]
Confirmed that our internal tests passed!

File: js/src/gitutil.ts (hunk 1 above)
Comment (wong-codaio) [reply js/src/gitutil.ts:30]: This is the error with stack trace:
```
Error: Error: spawn git ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
    at Object.action (/<internal path>/node_modules/.pnpm/braintrust@0.0.85/node_modules/braintrust/dist/index.js:7460:25)
    at PluginStore.exec (/<internal path>/node_modules/.pnpm/braintrust@0.0.85/node_modules/braintrust/dist/index.js:7485:25)
    at /<internal path>/node_modules/.pnpm/braintrust@0.0.85/node_modules/braintrust/dist/index.js:4956:51
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (/<internal path>/node_modules/.pnpm/braintrust@0.0.85/node_modules/braintrust/dist/index.js:4954:16)
    at GitExecutorChain.<anonymous> (/<internal path>/node_modules/.pnpm/braintrust@0.0.85/node_modules/braintrust/dist/index.js:4938:44)
    at Generator.next (<anonymous>)
    at fulfilled (/<internal path>/node_modules/.pnpm/braintrust@0.0.85/node_modules/braintrust/dist/index.js:3761:24)
```

Comment (wong-codaio) [reply js/src/gitutil.ts]: Removed warning: https://github.com/braintrustdata/braintrust-sdk/pull/83/commits/1511047026dfe0ecc39ea727f5d0342d73a6784e

**Comment by wong-codaio** [comment]
Would need some help pushing out a new version once it's merged 🙏

**Comment by ankrgyl** [comment]
Just published 0.0.87

**Pull request merged**